### PR TITLE
C28 backend setup

### PR DESCRIPTION
**Pull request by TheOneHoff**
## Change Summary
- Created the queue table for storage of data
- Implemented some basic routes for interfacing with database
- Example post request in the javascript

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [x] The pull request title has an issue number
- [x] The code works by "Smoke testing" (in other words, you have tested it your own and it works)
- [x] The code has documentation (either comments or in main documentation platform)

## Other Information
Due to my lack of understanding when it came to Alembic, in order for the database to work, you will have to delete your PostgreSQL data folder before running docker-compose up. You only need to delete it once, after that it should work

### CODE REVIEW

**Review by Alexkszzz — APPROVED** [review]

**Review by frinzekt — COMMENTED** [review]
(delayed review)

File: app/app/queue.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+# Add a new entry to the queue
+@queue.route('/add_entry', methods=["POST"])
+def add_to_queue():
+    new = Queue(studentName = request.headers['studentName'], 
+                        studentNumber = request.headers['studentNumber'], 
+                        unitCode = request.headers['unitCode'], 
+                        enquiry = request.headers['enquiry'],
+                        queue = request.headers['queue'],
+                        status = 'In Queue',
+                        enterQueueTime = datetime.now(pytz.timezone('Australia/Perth')))
```
[the remark this answers is not in the document]
Comment (TheOneHoff) [reply app/app/queue.py:21]: ty for the example vid. As I said i'm not too confident in my js, so i wan't sure how to use json correctly

File: app/app/queue.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    dest = request.headers['destination']
+    entry = db.session.query(Queue).filter(Queue.id == id).first()
+    src = entry.status
+    time = datetime.now(pytz.timezone('Australia/Perth'))
+
+    if src == 'Ended':
+        if dest == 'In Queue':
+            entry.exitSessionTime = Null
+        else:
+            raise ValueError("Invalid Destination")
```
[the remark this answers is not in the document]
Comment (TheOneHoff) [reply app/app/queue.py:40]: Oh interesting. I did not realize that was how it worked. Will keep it in mind for the future

File: app/app/static/custom/ourscript.js [hunk 3]
```diff
@@ -30,6 +31,17 @@ $('#addToQueueForm').submit(function (e) {
...
+	
+	fetch("add_entry", {
+        method: "POST",
+        headers: {
+            studentName: name,
+            studentNumber: id,
+            unitCode: unit,
+            enquiry: enquiry,
+            queue: team
+        }
```
[the remark this answers is not in the document]
Comment (TheOneHoff) [reply app/app/static/custom/ourscript.js:43]: Will look into the proper procedure for passing data between client and application. Definitely can improve in that area

**Pull request merged**